### ISSUE TITLE
[HOLD for payment 2023-09-04] Surface distance requests in App

### ISSUE THREAD

**Issue by neil-marcellini** [Expensify/App#22712]
Please follow the plan in the design doc: Surface distance requests [in App](https://docs.google.com/document/d/16Q9sE-991CZ4wUcalOs_U-qyEEza_C9tZwj6G5lHUp0/edit#heading=h.bojfyqj0rs1b).

- [ ] The display of the ReportPreview component will be [modified in Scan Receipt](https://docs.google.com/document/d/1Il9y1287B6tq0FB5ziXnhBcwn-Lw-n_rSmllZ-AyUVs/edit#bookmark=id.ph8hkjs6zejy) to match the style shown in the mockup, all we need to do differently is show the route image. 

- [ ] For distance request actions we will display the route receipt image in the [MoneyRequestAction](https://github.com/Expensify/App/blob/8bf3cbeddb1c0530f1a528b07517fb05c0cbc987/src/components/ReportActionItem/MoneyRequestAction.js#L93) component. 

- [ ] If it’s a distance request as determined by a new util function isDistanceRequest(transaction), we’ll subscribe to the transaction from MoneyRequestAction using withOnyx and the IOUTransactionID stored in action.message. Inside the function we’ll return whether transaction.type === “customUnit” and transaction.customUnit.name === “Distance”. This matches what we do on [OldDot](https://github.com/Expensify/Web-Expensify/blob/e2f13d08f7d/site/deprecated/lib_expense3.js#L1655-L1658). 

- [ ] For distance requests we’ll render the receipt image above the header as [described in the receipt doc](https://docs.google.com/document/d/1Il9y1287B6tq0FB5ziXnhBcwn-Lw-n_rSmllZ-AyUVs/edit#bookmark=id.mcbrkeq6ieqx). 

- [ ] We can create a small ReceiptUtils.getReceiptURL(filename) function to get the url from the receipt.filename. It will return the same URL format for receipts that we[ use for OldDot](https://github.com/Expensify/Web-Expensify/blob/117d1350320f2d58c836dc2eabef598e657034c7/lib/Receipt.php#L425). 

- [ ] We’ll set the title of the header to “Distance” if isDistanceRequest returns true. The amount will come from the transaction.amount or transaction.modifiedAmount. For distance requests the “Description” will come from the transaction.merchant instead of transaction.comment.comment as it does for manual requests, since that’s how it’s stored.

**Comment by marcaaron** [comment]
Looks like this is on HOLD and not overdue.

**Comment by marcaaron** [comment]
Still HOLD

**Comment by marcaaron** [comment]
This is on hold melv

**Comment by marcaaron** [comment]
hold

**Comment by marcaaron** [comment]
Why does Melvin not understand HOLD...?

**Comment by neil-marcellini** [comment]
I asked about that once. Apparently it's so that we don't forget about issues we put on HOLD. I'll drop it to a weekly for you.

**Comment by marcaaron** [comment]
Still on hold.

**Comment by marcaaron** [comment]
Filenames _should_ be coming back in the Auth response now 🎉 

@neil-marcellini is this still blocked on some other parts? e.g. that first step implies that we are waiting for something in Scan Receipt - but not too sure if that's done yet.

**Comment by neil-marcellini** [comment]
I think we're holding on this scan receipt issue https://github.com/Expensify/Expensify/issues/295245, and then also I'm not sure it makes sense to work on this in App until the requests are actually created on the backend, but maybe not?

Maybe we can insert a mock distance request in the dev DB and start working on this in App? Do you that's a good plan or we should wait until CreateDistanceRequest has been build across the stack.

**Comment by marcaaron** [comment]
I could go either way here. I think if we need anything from Scan Receipt then we should wait for that to avoid potential conflicts. After that - if there's no way to create a distance receipt yet, but we just want to set up the UI for them then it seems OK to start working on it.

**Comment by tgolen** [comment]
I've created [a draft PR](https://github.com/Expensify/App/pull/24556) with the changes for this. Moving onto testing it.

**Comment by tgolen** [comment]
Just waiting for payment as the title says. Not really overdue

**Comment by tgolen** [comment]
Oh, this was done internally and there will be no payment. This can be closed.
